### PR TITLE
allow disabling file detections by reading header

### PR DESCRIPTION
**Pull request by WeidiDeng**
**Description**
This pr solves the [issue](https://github.com/filebrowser/filebrowser/issues/1112) and provides a frontend to set the options.

:rotating_light: Before submitting your PR, please read [community](https://github.com/filebrowser/community), and indicate which issues (in any of the repos) are either fixed or closed by this PR. See [GitHub Help: Closing issues using keywords](https://help.github.com/articles/closing-issues-via-commit-messages/).

- [ ] DO make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [ ] DO make sure you are making a pull request against the **master branch** (left side). Also you should start *your branch* off *our master*.
- [ ] DO make sure that File Browser can be successfully built. See [builds](https://github.com/filebrowser/community/blob/master/builds.md) and [development](https://github.com/filebrowser/community/blob/master/development.md).
- [ ] DO make sure that related issues are opened in other repositories. I.e., the frontend, caddy plugins or the web page need to be updated accordingly.
- [ ] AVOID breaking the continuous integration build.

**Further comments**
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did, what alternatives you considered, etc.

:heart: Thank you!

### CODE REVIEW

**Comment by o1egl** [comment]
This is actually a duplicate of https://github.com/filebrowser/filebrowser/pull/1168

**Comment by WeidiDeng** [comment]
but that pr doesn't have the updated vue ui, no option exposed yet

**Comment by o1egl** [comment]
I don't think that we need any ui for that. Configuring this option per user doesn't make sense.

**Comment by WeidiDeng** [comment]
Perhaps this can be made in a global option, the one that uses commandline to set options. An array of relatives path to root, that skips type detection.

I tried the other [pr](https://github.com/filebrowser/filebrowser/pull/1168). Mostly it works fine, however I have a directory full of subtitle files (sass extension), the mime type detection fails, directory become very slow to show.

File: files/file.go [hunk 1]
```diff
@@ -261,9 +262,13 @@ func (i *FileInfo) readListing(checker rules.Checker) error {
...
-			err := file.detectType(true, false)
-			if err != nil {
-				return err
+			if detectDisabled {
+				file.Type = "blob"
+			} else {
+				err := file.detectType(true, false)
+				if err != nil {
+					return err
+				}
```
Comment (o1egl) [inline files/file.go]: We should only disable file detection by header. Simple ext based detection shouldn't be disabled.

**Review by o1egl — CHANGES_REQUESTED** [review]
In my opinion it doesn't make sense to disable it for specific paths only. It should be a global single boolean param.

File: files/file.go (hunk 1 above)
Comment (WeidiDeng) [reply files/file.go]: Now I incorporate the changes from [pr](https://github.com/filebrowser/filebrowser/pull/1168).

File: rules/rules.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ import (
 // Checker is a Rules checker.
 type Checker interface {
 	Check(path string) bool
+	ReadHeader() bool
```
Comment (o1egl) [inline rules/rules.go]: I don't think that it is the right place to add this method

File: files/file.go [hunk 3]
```diff
@@ -261,7 +269,7 @@ func (i *FileInfo) readListing(checker rules.Checker) error {
 		} else {
 			listing.NumFiles++
 
-			err := file.detectType(true, false)
+			err := file.detectType(true, false, checker.ReadHeader())
```
Comment (o1egl) [inline files/file.go]: Better would be to add a boolean param to the readListing func

**Review by o1egl — CHANGES_REQUESTED** [review]

**Comment by o1egl** [comment]
Instead of extending Checker interface you can add a boolean param to the FileOptions struct.

**Review by o1egl — APPROVED** [review]

**Pull request merged**